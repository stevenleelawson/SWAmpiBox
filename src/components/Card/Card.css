.card {
  border: 3px solid #fff;
  border-radius: 5px;
  height: 20rem;
  width: 15rem;
  margin: 5rem;

}

.name,
.species,
.homeworld,
.population {
  color: white;
}
