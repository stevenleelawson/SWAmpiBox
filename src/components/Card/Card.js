import React from 'react';
import './Card.css'

const Card = ({name, species, homeworld, population, toggleFavorites}) => {
  return (
    <div className='card'>
      <h1 className='name'>{name}</h1>
      <p className='species'>Species: {species}</p>
      <p className='homeworld'>Homeworld: {homeworld}</p>
      <p className='population'>Population: {population}</p>
      <button className='button' onClick={() => toggleFavorites({name, species, homeworld, population, category: 'people'})}>favorites</button>
    </div>
  )
}

export default Card;
