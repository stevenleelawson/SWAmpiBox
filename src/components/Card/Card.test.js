import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  })
})
