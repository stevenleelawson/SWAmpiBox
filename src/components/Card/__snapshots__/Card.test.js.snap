// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="name"
/>,
        <p
          className="species"
>
          Species: 
</p>,
        <p
          className="homeworld"
>
          Homeworld: 
</p>,
        <p
          className="population"
>
          Population: 
</p>,
        <button
          className="button"
          onClick={[Function]}
>
          favorites
</button>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "name",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species: ",
            undefined,
          ],
          "className": "species",
        },
        "ref": null,
        "rendered": Array [
          "Species: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld: ",
            undefined,
          ],
          "className": "homeworld",
        },
        "ref": null,
        "rendered": Array [
          "Homeworld: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population: ",
            undefined,
          ],
          "className": "population",
        },
        "ref": null,
        "rendered": Array [
          "Population: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorites",
          "className": "button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "favorites",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="name"
/>,
          <p
            className="species"
>
            Species: 
</p>,
          <p
            className="homeworld"
>
            Homeworld: 
</p>,
          <p
            className="population"
>
            Population: 
</p>,
          <button
            className="button"
            onClick={[Function]}
>
            favorites
</button>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "name",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              undefined,
            ],
            "className": "species",
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              undefined,
            ],
            "className": "homeworld",
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              undefined,
            ],
            "className": "population",
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorites",
            "className": "button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "favorites",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
